ConsoleApp1/Shop.cs

```diff
@@ -0,0 +1,317 @@
+            {
+                Console.WriteLine($"#{i+1} Product name: {products[i].Name} quantity: {products[i].Quantity}");
+            }
+        }
+
+        /// <summary>
+        /// Show all buyers registered in the shop
+        /// </summary>
+        public void ShowBuyersList()
+        {
+            Console.WriteLine("The List of Buyers");
+            for (int i = 0; i < buyers.Count; i++)
+            {
+                Console.WriteLine($"#{i + 1} Buyer name: {buyers[i].Name} Personal Info: {buyers[i].PersonalInfo}");
+            }
+        }
+
+        /// <summary>
+        /// Show info about all receipts collected in the shop
+        /// </summary>
+        public void ShowReceiptList()
+        {
+            Console.WriteLine("The List of Receipts");
+            foreach (var item in receipts)
+            {
+                Console.WriteLine($"Buyer Name: {item.Buyer.Name}");
+                Console.WriteLine($"Product: {item.SoldProduct.Name}");
+                Console.WriteLine($"Qantity: {item.Quantity}");
+                Console.WriteLine("- - - - - - - - - - - - - -");
+            }
+        }
+
+        private void ShopNavigation()
+        {
+            Console.WriteLine("0.Leave the shop");
+            Console.WriteLine("1.Register a new type of product to the shop");
+            Console.WriteLine("2.Add quantity to the existed product in the shop");
+            Console.WriteLine("3.Register a new buyer");
+            Console.WriteLine("4.Sell product");
+            Console.WriteLine("5.Show the list of products");
+            Console.WriteLine("6.Show the list of buyers");
+            Console.WriteLine("7.Show list of recipts");
+        }
+
+        /// <summary>
+        /// Add new product to the shop
+        /// </summary>
+        public void AddNewProduct()
+        {
+            Console.WriteLine("Provide the new name of the product");
+            string name = Console.ReadLine();
+
+            Console.WriteLine("Provide description of new product");
+            string description = Console.ReadLine();
+
+            Console.WriteLine("Provide price of new product");
+            int price;
+            int.TryParse(Console.ReadLine(), out price);
+            AddProduct(new Product(name, description, 0, price));
+        }
+
+        /// <summary>
+        /// Increase the quantity of the product in the shop
+        /// </summary>
+        public void AddProductQuantity()
+        {         
+            ShowProductsList();
+            Console.WriteLine("Choose the product to increase quantity.");
+            
+            int index;
+            
+            if (!int.TryParse(Console.ReadLine(), out index))
+            {
+                Console.WriteLine("Pls, choose the corect number");
+                Console.ReadKey();
+                return;
+            }
+
+            IProduct product = null;
+           
+            try
```

Comment:
better to check the range instead of exceptions

Reply:
done